blog/views.py

```diff
@@ -0,0 +1,46 @@
+from django.http import HttpResponse, HttpRequest, HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views import generic
+
+from blog.forms import CommentaryForm
+from blog.models import Post
+
+
+class PostListView(generic.ListView):
+    model = Post
+    queryset = Post.objects.select_related("owner").prefetch_related(
+        "commentaries"
+    )
+    paginate_by = 5
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+    def get_object(self, queryset=None) -> Post:
+        return Post.objects.prefetch_related("commentaries__user").get(
+            id=self.kwargs["pk"]
+        )
```

Comment:
In the `get_object` method, using `get` without handling exceptions like `Post.DoesNotExist` can lead to unhandled exceptions if the post ID does not exist. Consider using `get_object_or_404` to handle such cases gracefully.